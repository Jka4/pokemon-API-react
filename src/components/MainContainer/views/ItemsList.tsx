import React from 'react';

import { useSelector } from 'react-redux';

import Heading from 'components/MainContainer/views/Heading';
import PokemonCard from 'components/MainContainer/views/PokemonCard';

import { pokemonCardType } from 'types/index';

import styled from 'styled-components';
import { randomPokemonsSelector } from 'selectors';

const ItemsList: React.FC = () => {
  const randomPokemons = useSelector(randomPokemonsSelector);

  const haveData = randomPokemons.length !== 0;

  return (
    <>
      {haveData && <Heading />}

      <CardsContainer>
        {randomPokemons?.map((el: pokemonCardType, key: number) => (
          <PokemonCard pokeCard={el} key={key} />
        ))}
      </CardsContainer>
    </>
  );
};

const CardsContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding-bottom: 100px;
  width: 100%;
  height: auto;
`;

ItemsList.defaultProps = {
  randomPokemon: [],
};

export default ItemsList;
