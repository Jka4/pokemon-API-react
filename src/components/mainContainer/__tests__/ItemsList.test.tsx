import * as React from 'react';

import ItemsList from '../views/ItemsList';

import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter } from 'react-router-dom';

const mockStore = configureMockStore([thunk]);

describe('<ItemsList/> ', () => {
  it('renders correctly', () => {
    const pokemon = {
      id: 565,
      image: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/565.png',
      imageHQ:
        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/565.png',
      name: 'carracosta',
      order: 665,
      placeholderBase64:
        'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAOCAYAAAAfSC3RAAABfUlEQVQ4T2NkIBMwYtX3nwEs/p+BgYGREUxhAOwaiXAFVo0BDQoCG+b8+ZmS+014Tvm7JwRtXLWKgTksjOFvzSru0v//2E0Zmb6fbgn73j3zjDFrusnZ38gGoNjYsJ+BpcGR4U/mKqE6FfV3jU9vch/uC9MIYmA4+6ahgYGpoYHhH0wzXGNoaCjz6tWr/zLUBhh3a786oSD5nvH2Kxnm1//Pbu0Pe+cTuiqUeXXY6r8YGhtWabE1hF37VbJfJTlKVGGOwE+er8c+f+N6wb7rg5wlg2gYA8NfhwYHlgMNB/6ANMNsBNH/GRj8pNwbnkx3tXvmx8r6kYFBgIHh5RNFhhXrhVbc4+KpZJiw4wEDKKoYGf5DNDYwMDE0MPyTSbDTffL9v4ez0OtXGpJ/hB9xcqq9fct2//ZHQVYe3v8n73ft3Q1TiwgcqEl4oxBJDWo8gmx+ZszcIMX7/5nkZ0YpNd7/Bw4yMHx+9pnxrJTvX4aGBsxQJSKxoCghO8kBALL4iw/htJjEAAAAAElFTkSuQmCC',
      weight: 810,
    };

    const store = mockStore({ randomPokemon: [pokemon] });

    const wrapper = mount(
      <BrowserRouter basename="/">
        <Provider store={store}>
          <ItemsList />
        </Provider>
      </BrowserRouter>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
