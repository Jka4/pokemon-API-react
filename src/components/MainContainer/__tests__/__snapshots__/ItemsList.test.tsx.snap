// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemsList/>  renders correctly 1`] = `
<BrowserRouter
  basename="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <ItemsList
        randomPokemon={Array []}
      >
        <Heading>
          <styled.h2>
            <h2
              className="sc-bdfBwQ hqbntG"
            >
              <styled.ul>
                <ul
                  className="sc-gsTCUz bkxspn"
                >
                  <styled.li>
                    <li
                      className="sc-dlfnbm jMOKXL"
                    >
                      photo
                    </li>
                  </styled.li>
                  <styled.li>
                    <li
                      className="sc-dlfnbm jMOKXL"
                    >
                      name
                    </li>
                  </styled.li>
                  <styled.li>
                    <li
                      className="sc-dlfnbm jMOKXL"
                    >
                      order
                    </li>
                  </styled.li>
                  <styled.li>
                    <li
                      className="sc-dlfnbm jMOKXL"
                    >
                      exp
                    </li>
                  </styled.li>
                </ul>
              </styled.ul>
            </h2>
          </styled.h2>
        </Heading>
        <styled.div>
          <div
            className="sc-kEjbxe irGWPW"
          >
            <PokemonCard
              key="0"
              pokeCard={
                Object {
                  "id": 565,
                  "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/565.png",
                  "imageHQ": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/565.png",
                  "name": "carracosta",
                  "order": 665,
                  "placeholderBase64": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAOCAYAAAAfSC3RAAABfUlEQVQ4T2NkIBMwYtX3nwEs/p+BgYGREUxhAOwaiXAFVo0BDQoCG+b8+ZmS+014Tvm7JwRtXLWKgTksjOFvzSru0v//2E0Zmb6fbgn73j3zjDFrusnZ38gGoNjYsJ+BpcGR4U/mKqE6FfV3jU9vch/uC9MIYmA4+6ahgYGpoYHhH0wzXGNoaCjz6tWr/zLUBhh3a786oSD5nvH2Kxnm1//Pbu0Pe+cTuiqUeXXY6r8YGhtWabE1hF37VbJfJTlKVGGOwE+er8c+f+N6wb7rg5wlg2gYA8NfhwYHlgMNB/6ANMNsBNH/GRj8pNwbnkx3tXvmx8r6kYFBgIHh5RNFhhXrhVbc4+KpZJiw4wEDKKoYGf5DNDYwMDE0MPyTSbDTffL9v4ez0OtXGpJ/hB9xcqq9fct2//ZHQVYe3v8n73ft3Q1TiwgcqEl4oxBJDWo8gmx+ZszcIMX7/5nkZ0YpNd7/Bw4yMHx+9pnxrJTvX4aGBsxQJSKxoCghO8kBALL4iw/htJjEAAAAAElFTkSuQmCC",
                  "weight": 810,
                }
              }
            >
              <styled.div>
                <div
                  className="sc-iBPRYJ jzmLdG"
                >
                  <NavLink
                    to="/detailedPage/carracosta"
                  >
                    <Link
                      aria-current={null}
                      to={
                        Object {
                          "hash": "",
                          "pathname": "/detailedPage/carracosta",
                          "search": "",
                          "state": null,
                        }
                      }
                    >
                      <LinkAnchor
                        aria-current={null}
                        href="/detailedPage/carracosta"
                        navigate={[Function]}
                      >
                        <a
                          aria-current={null}
                          href="/detailedPage/carracosta"
                          onClick={[Function]}
                        >
                          <Styled(styled.div)>
                            <div
                              className="sc-iBPRYJ sc-fubCfw jzmLdG kSIUcg"
                            >
                              <styled.div>
                                <div
                                  className="sc-pFZIQ iCnbvo"
                                >
                                  <ImageContainer
                                    fallback={
                                      <img
                                        alt=""
                                        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAOCAYAAAAfSC3RAAABfUlEQVQ4T2NkIBMwYtX3nwEs/p+BgYGREUxhAOwaiXAFVo0BDQoCG+b8+ZmS+014Tvm7JwRtXLWKgTksjOFvzSru0v//2E0Zmb6fbgn73j3zjDFrusnZ38gGoNjYsJ+BpcGR4U/mKqE6FfV3jU9vch/uC9MIYmA4+6ahgYGpoYHhH0wzXGNoaCjz6tWr/zLUBhh3a786oSD5nvH2Kxnm1//Pbu0Pe+cTuiqUeXXY6r8YGhtWabE1hF37VbJfJTlKVGGOwE+er8c+f+N6wb7rg5wlg2gYA8NfhwYHlgMNB/6ANMNsBNH/GRj8pNwbnkx3tXvmx8r6kYFBgIHh5RNFhhXrhVbc4+KpZJiw4wEDKKoYGf5DNDYwMDE0MPyTSbDTffL9v4ez0OtXGpJ/hB9xcqq9fct2//ZHQVYe3v8n73ft3Q1TiwgcqEl4oxBJDWo8gmx+ZszcIMX7/5nkZ0YpNd7/Bw4yMHx+9pnxrJTvX4aGBsxQJSKxoCghO8kBALL4iw/htJjEAAAAAElFTkSuQmCC"
                                      />
                                    }
                                    url="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/565.png"
                                  >
                                    <ErrorBoundary>
                                      <styled.img
                                        isLoaded={false}
                                        src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/565.png"
                                      >
                                        <img
                                          className="sc-gKsewC eLDCSu"
                                          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/565.png"
                                        />
                                      </styled.img>
                                      <styled.div
                                        isLoaded={false}
                                      >
                                        <div
                                          className="sc-jSgupP hKlLsH"
                                        >
                                          <img
                                            alt=""
                                            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAOCAYAAAAfSC3RAAABfUlEQVQ4T2NkIBMwYtX3nwEs/p+BgYGREUxhAOwaiXAFVo0BDQoCG+b8+ZmS+014Tvm7JwRtXLWKgTksjOFvzSru0v//2E0Zmb6fbgn73j3zjDFrusnZ38gGoNjYsJ+BpcGR4U/mKqE6FfV3jU9vch/uC9MIYmA4+6ahgYGpoYHhH0wzXGNoaCjz6tWr/zLUBhh3a786oSD5nvH2Kxnm1//Pbu0Pe+cTuiqUeXXY6r8YGhtWabE1hF37VbJfJTlKVGGOwE+er8c+f+N6wb7rg5wlg2gYA8NfhwYHlgMNB/6ANMNsBNH/GRj8pNwbnkx3tXvmx8r6kYFBgIHh5RNFhhXrhVbc4+KpZJiw4wEDKKoYGf5DNDYwMDE0MPyTSbDTffL9v4ez0OtXGpJ/hB9xcqq9fct2//ZHQVYe3v8n73ft3Q1TiwgcqEl4oxBJDWo8gmx+ZszcIMX7/5nkZ0YpNd7/Bw4yMHx+9pnxrJTvX4aGBsxQJSKxoCghO8kBALL4iw/htJjEAAAAAElFTkSuQmCC"
                                          />
                                        </div>
                                      </styled.div>
                                    </ErrorBoundary>
                                  </ImageContainer>
                                </div>
                              </styled.div>
                              <styled.span>
                                <span
                                  className="sc-jrAGrp iCHhJL"
                                >
                                  carracosta
                                </span>
                              </styled.span>
                              <styled.span>
                                <span
                                  className="sc-jrAGrp iCHhJL"
                                >
                                  665
                                </span>
                              </styled.span>
                              <styled.span>
                                <span
                                  className="sc-jrAGrp iCHhJL"
                                />
                              </styled.span>
                            </div>
                          </Styled(styled.div)>
                        </a>
                      </LinkAnchor>
                    </Link>
                  </NavLink>
                </div>
              </styled.div>
            </PokemonCard>
          </div>
        </styled.div>
      </ItemsList>
    </Provider>
  </Router>
</BrowserRouter>
`;
